URI scheme Pattern replace error

Hi,

When setting the URI with "http://10.0.3.2:8888" and trying to connect to the server I get the below exception:

``` java
    java.io.IOException: java.lang.IllegalArgumentException: The URI scheme, of the URI wsnull://10.0.3.2:8888, must be equal (ignoring case) to 'http', 'https', 'ws', or 'wss'
            at org.atmosphere.wasync.transport.WebSocketTransport.connectFutureException(WebSocketTransport.java:260) ~[na:0.0]
            at org.atmosphere.wasync.transport.WebSocketTransport.onFailure(WebSocketTransport.java:295) ~[na:0.0]
            at org.atmosphere.wasync.transport.WebSocketTransport.onThrowable(WebSocketTransport.java:114) ~[na:0.0]
            at org.atmosphere.wasync.impl.DefaultSocket.connect(DefaultSocket.java:177) ~[na:0.0]
            at org.atmosphere.wasync.impl.DefaultSocket.open(DefaultSocket.java:126) ~[na:0.0]
            at org.atmosphere.wasync.impl.DefaultSocket.open(DefaultSocket.java:109) ~[na:0.0]
            at com.app.client.SocketIOTest.wAsyncSocket(SocketIOTest.java:152) ~[na:0.0]
            at com.app.client.SocketIOTest.onCreate(SocketIOTest.java:58) ~[na:0.0]
            at android.app.Activity.performCreate(Activity.java:5231) ~[na:0.0]
            at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1087) ~[na:0.0]
            at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2159) ~[na:0.0]
            at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2245) ~[na:0.0]
            at android.app.ActivityThread.access$800(ActivityThread.java:135) ~[na:0.0]
            at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1196) ~[na:0.0]
            at android.os.Handler.dispatchMessage(Handler.java:102) ~[na:0.0]
            at android.os.Looper.loop(Looper.java:136) ~[na:0.0]
            at android.app.ActivityThread.main(ActivityThread.java:5017) ~[na:0.0]
            at java.lang.reflect.Method.invokeNative(Native Method) ~[na:0.0]
            at java.lang.reflect.Method.invoke(Method.java:515) ~[na:0.0]
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:779) ~[na:0.0]
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:595) ~[na:0.0]
            at dalvik.system.NativeStart.main(Native Method) ~[na:0.0]
     Caused by: java.lang.IllegalArgumentException: The URI scheme, of the URI wsnull://10.0.3.2:8888, must be equal (ignoring case) to 'http', 'https', 'ws', or 'wss'
            at com.ning.http.util.AsyncHttpProviderUtils.validateSupportedScheme(AsyncHttpProviderUtils.java:55) ~[na:0.0]
            at com.ning.http.client.RequestBuilderBase$RequestImpl.toURI(RequestBuilderBase.java:175) ~[na:0.0]
            at com.ning.http.client.RequestBuilderBase$RequestImpl.getURI(RequestBuilderBase.java:158) ~[na:0.0]
            at com.ning.http.client.RequestBuilderBase$RequestImpl.getUrl(RequestBuilderBase.java:144) ~[na:0.0]
            at com.ning.http.client.providers.netty.NettyAsyncHttpProvider.doConnect(NettyAsyncHttpProvider.java:988) ~[na:0.0]
            at com.ning.http.client.providers.netty.NettyAsyncHttpProvider.execute(NettyAsyncHttpProvider.java:933) ~[na:0.0]
            at com.ning.http.client.AsyncHttpClient.executeRequest(AsyncHttpClient.java:524) ~[na:0.0]
            at com.ning.http.client.AsyncHttpClient$BoundRequestBuilder.execute(AsyncHttpClient.java:236) ~[na:0.0]
            at org.atmosphere.wasync.impl.DefaultSocket.connect(DefaultSocket.java:154) ~[na:0.0]
```

The "wsnull" is due to the pattern given in DefaultSocket.java#connect method

```
r.setUrl(request.uri().replaceFirst ("^http(s)?://", "ws$1://"));
```

Is this the right pattern?

--- a/wasync/src/main/java/org/atmosphere/wasync/impl/DefaultSocket.java
+++ b/wasync/src/main/java/org/atmosphere/wasync/impl/DefaultSocket.java
@@ -149,7 +149,8 @@ public class DefaultSocket implements Socket {
 
         try {
             if (transportInUse.name().equals(Request.TRANSPORT.WEBSOCKET)) {
-                r.setUrl(request.uri().replaceFirst ("^http(s)?://", "ws$1://"));
+				// Android uses the 'ICU’s RegEx engine' and hence will replace a regex captured group with null if not found instead of an empty string. Below code fixes that
+                r.setUrl(request.uri().replaceFirst ("^http(s)?://", "ws$1://").replaceFirst("null", ""));
                 try {
                     transportInUse.future(new FutureProxy<ListenableFuture>(this,
                             options.runtime().prepareRequest(r.build()).execute((AsyncHandler<WebSocket>) transportInUse)));

